framework/test/unit/application.spec.ts

```diff
@@ -194,6 +232,29 @@ describe('Application', () => {
 		});
 	});
 
+	describe('#registerModule', () => {
+		it('should throw error when module with same name is already registered', () => {
+			// Arrange
+			const { app } = Application.defaultApplication(config);
+			class MyPlugin extends TestPlugin {
+				public get name() {
+					return 'my-plugin';
+				}
+			}
+			class MyModule extends TestModule {
+				public get name() {
+					return 'my-plugin';
+				}
+			}
```

Comment:
To make the test suite more readable and make it more obvious why it introduces new classes, it is always a good idea to replace generic names `MyModule` and `MyPlugin` with more specific names, such as `DuplicatedNameModule` and `DuplicatedNamePlugin` and set their names to something specific such as `duplicated`.